tests/tensor_search/test_image_download_headers.py

```diff
@@ -0,0 +1,161 @@
+"""
+Module for testing image download headers.
+"""
+import unittest.mock
+# we are renaming get to prevent inf. recursion while mocking get():
+from requests import get as requests_get
+from marqo.tensor_search.models.api_models import BulkSearchQuery
+from unittest import mock
+import requests
+from marqo.s2_inference.clip_utils import load_image_from_path
+from marqo.tensor_search.enums import IndexSettingsField
+from marqo.errors import IndexNotFoundError
+from marqo.tensor_search import tensor_search
+from tests.marqo_test import MarqoTestCase
+
+
+class TestImageDownloadHeaders(MarqoTestCase):
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super(TestImageDownloadHeaders, cls).setUpClass()
+        cls.generic_header = {"Content-type": "application/json"}
+        cls.index_name_1 = "my-test-index-1"
+        cls.real_img_url = 'https://raw.githubusercontent.com/marqo-ai/marqo-api-tests/mainline/assets/ai_hippo_realistic.png'
+
+    def setUp(self) -> None:
+        self.endpoint = self.authorized_url
+        try:
+            tensor_search.delete_index(config=self.config, index_name=self.index_name_1)
+        except IndexNotFoundError:
+            pass
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.index_name_1 = "my-test-index-1"
+        try:
+            tensor_search.delete_index(config=cls.config, index_name=cls.index_name_1)
+        except IndexNotFoundError:
+            pass
+
+    def image_index_settings(self) -> dict:
+        return {
+                IndexSettingsField.index_defaults: {
+                    IndexSettingsField.model: "ViT-B/32",
+                    IndexSettingsField.treat_urls_and_pointers_as_images: True,
+                }
+            }
+
+    def test_img_download_search(self):
```

Comment:
No entirely related, but this is a good example of a reason to move network calls from within to a standalone class/api (e.g. even using the Opensearch API). mocking would be easier and could be done with method decorators without being concerned about, for example, multiple calls to `requests.get` to mock.